app/controllers/users/verify_account_controller.rb

```diff
@@ -36,10 +34,14 @@ def params_otp
     end
 
     def confirm_verification_needed
-      return if current_user.decorate.pending_profile_requires_verification?
+      return if user_has_pending_profile? && FeatureManagement.reveal_usps_flow?
```

Comment:
We can eliminate the need of this additional condition by restricting at the route level.